table_engine/src/table.rs

```diff
@@ -350,6 +387,37 @@ pub struct ReadRequest {
     pub order: ReadOrder,
 }
 
+impl TryFrom<proto::remote_engine::TableReadRequest> for ReadRequest {
+    type Error = Error;
+
+    fn try_from(pb: proto::remote_engine::TableReadRequest) -> Result<Self> {
+        Ok(Self {
+            request_id: RequestId::next_id(),
+            opts: pb.opts.context(EmptyReadOptions)?.into(),
+            projected_schema: pb
+                .projected_schema
+                .context(EmptyProjectedSchema)?
+                .try_into()
+                .map_err(|e| Box::new(e) as _)
+                .context(ConvertProjectedSchema)?,
+            predicate: Arc::new(
+                pb.predicate
+                    .context(EmptyPredicate)?
+                    .try_into()
+                    .map_err(|e| Box::new(e) as _)
+                    .context(ConvertPredicate)?,
+            ),
+            order: if pb.order == proto::remote_engine::ReadOrder::Asc as i32 {
+                ReadOrder::Asc
+            } else if pb.order == proto::remote_engine::ReadOrder::Desc as i32 {
+                ReadOrder::Desc
+            } else {
+                ReadOrder::None
+            },
```

Comment:
Shall split up these fields' initialization? It looks crowded to squeeze all of them in the `Self` initialization.